src/twisted/conch/scripts/ckeygen.py

```diff
@@ -202,10 +203,22 @@ def _defaultPrivateKeySubtype(keyType):
         return "PEM"
 
 
-def printFingerprint(options):
+def _getKeyOrDefault(options):
+    """
+    If C{options["filename"]} is None, prompt the user to enter a path
+    or attempt to set it to .ssh/id_rsa
+    """
     if not options["filename"]:
         filename = os.path.expanduser("~/.ssh/id_rsa")
-        options["filename"] = input("Enter file in which the key is (%s): " % filename)
+        if platform.system() == "Windows":
+            filename = os.path.expandvars(R"%HOMEPATH %\.ssh\id_rsa")
+        options["filename"] = (
+            input("Enter file in which the key is (%s): " % filename) or filename
+        )
+
+
+def printFingerprint(options):
+    _getKeyOrDefault(options)
```

Comment:
I suggest that `_getKeyOrDefault` avoids mutating `options` and instead returns the value for `filename`.  Then `printFingerprint` can just use that return value directly rather than fishing around inside the `options` object.

Pure functions are better than functions with side-effects and immutable state is better than mutable state.

Twisted has a _lot_ of mutable state all over so I don't think the current `_getKeyOrDefault` behavior is particularly out of place - but almost all of those places in Twisted that work this way are mistakes that, ideally, would gradually be fixed over time.  We can just make this new API avoid this mistake in the first place. :)

Reply:
I think I have also commented about returning a value, instead of mutating the input arguments.

so I guess that this is required for approving this PR

Reply:
Updated so that _getKeyOrDefault doesn't mutate options.